lib/gst/player/gstplayer.c

```diff
@@ -1370,6 +1376,8 @@ state_changed_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
           gst_player_seek_internal_locked (self);
         } else {
           GST_DEBUG_OBJECT (self, "Seek finished");
+          g_signal_emit (self, signals[SIGNAL_SEEK_DONE], 0,
+              gst_player_get_position (self));
```

Comment:
See how the other signals are emitted, you need to dispatch it to the application's main context if that was requested by the application.